torch/nn/functional.py

```diff
@@ -1827,8 +1827,22 @@ def multilabel_soft_margin_loss(input, target, weight=None, size_average=None,
     """
     if size_average is not None or reduce is not None:
         reduction = _Reduction.legacy_get_string(size_average, reduce)
-    input = torch.sigmoid(input)
-    return binary_cross_entropy(input, target, weight, None, None, reduction)
+
+    loss = -(target * logsigmoid(input) + (1 - target) * logsigmoid(-input))
```

Comment:
this could probably use `torch.lerp` (yet AFAIK it fails export to ONNX and has NaN problems: https://github.com/pytorch/pytorch/issues/71701)